app/src/main/java/org/scesi/cappuchino/ui/utils/SearchBar.kt

```diff
@@ -0,0 +1,158 @@
+import androidx.compose.runtime.mutableStateOf
+import androidx.compose.runtime.remember
+import androidx.compose.runtime.setValue
+import androidx.compose.ui.Alignment
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.draw.clip
+import androidx.compose.ui.graphics.Color
+import androidx.compose.ui.res.dimensionResource
+import androidx.compose.ui.res.painterResource
+import androidx.compose.ui.res.stringResource
+import androidx.compose.ui.tooling.preview.Preview
+import androidx.compose.ui.unit.dp
+import androidx.compose.ui.unit.sp
+import org.scesi.cappuchino.R
+
+sealed class SearchCategory(
+    var searchCriteria: String
+) {
+    data class Subject(
+        val name: String
+    ) : SearchCategory(name)
+
+}
+
+@OptIn(ExperimentalMaterial3Api::class)
+@Composable
+fun SearchBar(career: List<SearchCategory>, onClick: () -> Unit) {
+
+    var searchText by remember { mutableStateOf("") }
+    val filteredCarreras = career.filter { carrera ->
+        carrera.searchCriteria.contains(searchText, ignoreCase = true)
+    }
+
+    OutlinedTextField(
+        value = searchText,
+        onValueChange = { searchText = it },
+        placeholder = {
+            Text(
+                text = stringResource(R.string.text_searchbar),
+                fontSize = dimensionResource(id = R.dimen.text_size_small).value.sp
+            )
+        },
+        trailingIcon = {
+            Icon(
+                painter = painterResource(id = R.drawable.lupaicon),
+                contentDescription = stringResource(R.string.Search),
+                modifier = Modifier.size(24.dp),
+                tint = Color.Gray
+            )
+        },
+        modifier = Modifier
+            .fillMaxWidth()
+            .height(dimensionResource(id = R.dimen.margin_padding_size_xxxlarge))
+            .padding(horizontal = dimensionResource(id = R.dimen.margin_padding_size_smallv1))
+            .clip(RoundedCornerShape(dimensionResource(id = R.dimen.margin_padding_size_small)))
+            .clickable { onClick() },
+        colors = TextFieldDefaults.outlinedTextFieldColors(
+            focusedBorderColor = Color.Gray,
+            unfocusedBorderColor = Color.Gray
+        ),
+        shape = RoundedCornerShape(dimensionResource(id = R.dimen.margin_padding_size_small))
+    )
+    Spacer(modifier = Modifier.height(16.dp))
+
+    Column(
+        modifier = Modifier
+            .fillMaxSize()
+            .verticalScroll(rememberScrollState())
+    ){
+        if (filteredCarreras.isEmpty()) {
+            Text(
+                text = stringResource(id = R.string.not_found),
+                color = Color.Gray,
+                modifier = Modifier.align(Alignment.CenterHorizontally)
+            )
+        } else {
+            filteredCarreras.forEach { carrera ->
+                TextBoxSearch(text = carrera.searchCriteria, onClick = {
+                })
+            }
+        }
+    }
+}
+//@Preview(showBackground = true)
```

Comment:
display always a preview, uncomment this , and previews can be placed at the bottom of the file